core/domain/exp_services.py

```diff
@@ -1864,24 +1865,25 @@ def is_version_of_draft_valid(exp_id, version):
 def get_user_exploration_data(
         user_id, exploration_id, apply_draft=False, version=None):
     """Returns a description of the given exploration."""
+    exp_user_data = user_models.ExplorationUserDataModel.get(
+        user_id, exploration_id)
     if apply_draft:
-        exploration = get_exp_with_draft_applied(exploration_id, user_id)
+        is_valid_draft_version, exploration = (
+            get_exp_with_draft_applied(exploration_id, user_id))
     else:
         exploration = get_exploration_by_id(exploration_id, version=version)
+        is_valid_draft_version = (
```

Comment:
In the apply_draft=False case, shouldn't we be refraining from doing anything with the draft at all?

Reply:
Right now we just pass this to the front end and I'm not sure if the frontend does anything with is_valid_draft_version if we are just rendering a plain exploration. The code here just adheres to the same behavior. what do you think?

Reply:
I think it needs more analysis. In what situations is this function called with apply_draft=False? We need to understand the context in order to figure out what to do with is_valid_draft_version, I think.

Reply:
Looks like apply_draft=False is called on the put methods after update_exploration to load the latest version of the exploration.
is_version_of_draft_valid is read to decide to show a version mismatch modal. I think we can just set to true in the apply_draft=False case?

Reply:
Hm interesting. Yeah, I guess it would be trivially true in that case. Though what would be better is if the frontend didn't look at is_version_of_draft_valid at all in that case, since it doesn't really make sense to reference the draft when the put() call is explicitly avoiding it.

(Though by put() do you mean get()? Seems a bit odd to me that we're "loading data" using a put().)

Reply:
I was referring to the put() where we load the exploration right after the update
```
        try:
            exp_services.update_exploration(
                self.user_id, exploration_id, change_list, commit_message,
                is_by_voice_artist=True)
        except utils.ValidationError as e:
            raise self.InvalidInputException(e)

        exploration_data = exp_services.get_user_exploration_data(
            self.user_id, exploration_id)
```

Reply:
Mm. Can we set it to None, will that work? It's what was used originally, in line 1884 of the previous version of the code.

Reply:
I made a change here such that the only update to is_valid_draft_version happens if we were able to apply the draft to the exploration (in which case we know the draft version is valid).